src/walletSdk/Anchor/index.ts

```diff
@@ -195,6 +218,183 @@ export class Anchor {
     }
   }
 
+  /**
+  * Watch all transactions returned from a transfer server. When new or
+  * updated transactions come in, run an `onMessage` callback.
+  *
+  * On initial load, it'll return ALL pending transactions via onMessage.
+  * Subsequent messages will be any one of these events:
+  *  * Any new transaction appears
+  *  * Any of the initial pending transactions change any state
+  *
+  * You may also provide an array of transaction ids, `watchlist`, and this
+  * watcher will always react to transactions whose ids are in the watchlist.
+  */
+  watchAllTransactions(params: {
+    authToken: string;
+    assetCode: string;
+    onMessage: (transaction) => void;
+    onError: (error: any) => void;
+    watchlist?: string[];
+    timeout?: number;
+    isRetry?: boolean;
+    noOlderThan?: string;
+    kind?: string;
+    lang?: string;
+  }): WatcherResponse {
+    const {
+      authToken,
+      assetCode,
+      onMessage,
+      onError,
+      watchlist = [],
+      timeout = 5000,
+      isRetry = false,
+      ...otherParams
+    } = params;
+    
+    // make an object map out of watchlist
+    const watchlistMap: any = watchlist.reduce(
+      (memo: any, id: string) => ({ ...memo, [id]: true }),
+      {},
+    );
+
+    // make sure to initiate registries for the given asset code
+    // to prevent 'Cannot read properties of undefined' errors
+    if(!this._transactionsRegistry[assetCode]) {
+      this._transactionsRegistry[assetCode] = {};
+    }
+    if(!this._transactionsIgnoredRegistry[assetCode]) {
+      this._transactionsIgnoredRegistry[assetCode] = {};
+    }
+
+    // if it's a first run, drop it in the registry for the given asset code
+    if (!isRetry) {
+      this._watchAllTransactionsRegistry[assetCode] = true;
+    }
+
+    this.getTransactionsForAsset({ authToken, assetCode, ...(otherParams || {}) })
+      .then((transactions: any[]) => { // TOOD - replace with Transaction[] type
```

Comment:
instead let's use async/await to be consistent with other code. Or is there a reason this needs to be .then instead?
```suggestion
      .then((transactions: any[]) => { // TOOD - replace with Transaction[] type
```

Reply:
@acharb Good point.

Since this is a watcher functionality we have the option to call it `without awaiting anything`, like this:
```
      anchor.watchAllTransactions({
        authToken, 
        assetCode: "SRT",
        lang: "en-US",
        onMessage,
        onError,
      });
```
because we are only interested in the async values returned by `onMessage` and `onError` callbacks whenever they have something

But, in case we add any `await` inside the upper scope of the function body then `watchAllTransactions` would become an `async function` and devs would end up calling it like this:
```
      await anchor.watchAllTransactions({
        authToken, 
        assetCode: "SRT",
        lang: "en-US",
        onMessage,
        onError,
      });
```
^ in this case, the code would await for the `getTransactionsForAsset` API call to complete to only then proceed to the next line which would reduce performance a little bit

I think both ways could work, it's just that not making it an async function seems a bit more convenient and improves performance

How does that sound to you?

Reply:
hmm I see your point. I guess this way we're enforcing the watching code to happen async, where as if we make `watchAllTransactions` an `async func` then it would give devs the option to use async or not (by including `await` or not), but may confuse them to always think they need to await.

I don't feel strongly about changing, sgtm to keep!